File: app/src/main/java/org/deiverbum/app/data/entity/LHHymnWithAll.java
package org.deiverbum.app.data.entity;

import androidx.room.Embedded;
import androidx.room.Relation;

import org.deiverbum.app.model.Himno;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2022.2
 */
public class LHHymnWithAll {
    @Embedded
    public LHHymnJoinEntity himnoJoin;
    @Relation(
            parentColumn = "hymnFK",
            entityColumn = "hymnID",
            entity = LHHymnEntity.class
    )
    public LHHymnEntity himno;

    public Himno getDomainModel(){
        Himno dm=new Himno();
        dm.setTexto(himno.getHimno());
        return dm;
    }
}

File: app/src/main/java/org/deiverbum/app/core/database/model/relation/LHHymnWithAll.kt
package org.deiverbum.app.core.database.model.relation

import androidx.room.Embedded
import androidx.room.Relation
import org.deiverbum.app.core.database.model.entity.LHHymnEntity
import org.deiverbum.app.core.database.model.entity.LHHymnJoinEntity

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
data class LHHymnWithAll(
    @Embedded
    var himnoJoin: LHHymnJoinEntity,

    @Relation(parentColumn = "hymnFK", entityColumn = "hymnID", entity = LHHymnEntity::class)
    var himno: LHHymnEntity
)